hpx/util/detail/pack_traversal_impl.hpp

```diff
@@ -580,6 +644,13 @@ namespace util {
                     std::forward<T>(element));
             }
 
+            /// Boxes the given values into an according tuple
+            template <typename... T>
+            auto box(T&&... args) -> tuple<T...>
+            {
+                return tuple<T...>{std::forward<T>(args)...};
```

Comment:
Actually no, The use of `tuple<T...>{ ... }` is the result of a earlier compilatio issue with preserving l-value references across mapping, for instance this example would fail:

```cpp
    // Multiple args: Make it possible to pass move only objects in
    // as reference, while returning those as reference.
    {
        std::unique_ptr<int> ptr1(new int(6));
        std::unique_ptr<int> ptr2(new int(7));

        hpx::util::tuple<std::unique_ptr<int> const&,
            std::unique_ptr<int> const&>
            ref = map_pack(
                [](std::unique_ptr<int> const& ref)
                    -> std::unique_ptr<int> const& {
                    // ...
                    return ref;
                },
                ptr1, ptr2);

        HPX_TEST_EQ((*get<0>(ref)), 6);
        HPX_TEST_EQ((*get<1>(ref)), 7);
        *ptr1 = 1;
        *ptr2 = 2;
        HPX_TEST_EQ((*get<0>(ref)), 1);
        HPX_TEST_EQ((*get<1>(ref)), 2);
    }
```